src/sas/qtgui/Utilities/FileConverter.py

```diff
@@ -0,0 +1,487 @@
+        """
+        detector = Detector()
+        detector.name = self.txtMD_Name.text()
+        detector.distance = Utilities.toFloat(self.txtMD_Distance.text())
+        detector.offset = Vector(x=Utilities.toFloat(self.txtMD_OffsetX.text()),
+                                 y=Utilities.toFloat(self.txtMD_OffsetY.text()))
+        detector.orientation = Vector(x=Utilities.toFloat(self.txtMD_OrientRoll.text()),
+                                      y=Utilities.toFloat(self.txtMD_OrientPitch.text()),
+                                      z=Utilities.toFloat(self.txtMD_OrientYaw.text()))
+        detector.beam_center = Vector(x=Utilities.toFloat(self.txtMD_BeamX.text()),
+                                      y=Utilities.toFloat(self.txtMD_BeamY.text()))
+        detector.pixel_size = Vector(x=Utilities.toFloat(self.txtMD_PixelX.text()),
+                                     y=Utilities.toFloat(self.txtMD_PixelY.text()))
+        detector.slit_length = Utilities.toFloat(self.txtMD_Distance.text())
+        return detector
+
+    def getSampleMetadata(self):
+        """
+        Read the sample metadata fields and put them in the dictionary
+        """
+        sample = Sample()
+        sample.name = self.txtMSa_Name.text()
+        sample.beam_size = Vector(x=Utilities.toFloat(self.txtMSo_BeamSizeX.text()),
+                                  y=Utilities.toFloat(self.txtMSo_BeamSizeY.text()))
+        sample.beam_shape = self.txtMSo_BeamShape.text()
+        sample.wavelength = Utilities.toFloat(self.txtMSo_BeamWavelength.text())
+        sample.wavelength_min = Utilities.toFloat(self.txtMSo_MinWavelength.text())
+        sample.wavelength_max = Utilities.toFloat(self.txtMSo_MaxWavelength.text())
+        sample.wavelength_spread = Utilities.toFloat(self.txtMSo_Spread.text())
+        return sample
+
+    def getSourceMetadata(self):
+        """
+        Read the source metadata fields and put them in the dictionary
+        """
+        source = Source()
+        # radiation is on the front panel
+        source.radiation = self.cbRadiation.currentText().lower()
+        # the rest is in the 'Source' tab of the Metadata tab
+        source.name = self.txtMSo_Name.text()
+        source.thickness = Utilities.toFloat(self.txtMSa_Thickness.text())
+        source.transmission = Utilities.toFloat(self.txtMSa_Transmission.text())
+        source.temperature = Utilities.toFloat(self.txtMSa_Temperature.text())
+        source.position = Vector(x=Utilities.toFloat(self.txtMSa_PositionX.text()),
+                                 y=Utilities.toFloat(self.txtMSa_PositionY.text()))
+        source.orientation = Vector(x=Utilities.toFloat(self.txtMSa_OrientR.text()),
+                                    y=Utilities.toFloat(self.txtMSa_OrientP.text()),
+                                    z=Utilities.toFloat(self.txtMSa_OrientY.text()))
+
+        source.details = self.txtMSa_Details.toPlainText()
```

Comment:
Like my other comment, shouldn't all of this be in getSampleMetadata()?

Reply:
oh my.. embarrassing mistake ;)